sequencing/ATACSeq/jobSubmission/1_batchRunPreAnalysis.sh

```diff
@@ -0,0 +1,159 @@
+fi
+
+## ============ ##
+##    STEPS     ##
+## ============ ##
+
+## check if file containing list of sample IDs exists and if so:
+if test -f ${METADIR}/samples.txt;
+then 
+    ## create an array from the file
+    mapfile -t SAMPLEIDS < ${METADIR}/samples.txt 
+    echo "Number of sample IDs found:"" ""${#SAMPLEIDS[@]}"""
+    
+    ## Sample(s) specified in by array number(s) from command line
+    sampleID=${SAMPLEIDS[${SLURM_ARRAY_TASK_ID}]}
+    dirs=( ${RAWDATADIR} ${RAWDATADIR_B1} ${RAWDATADIR_B2} )
+    #toProcess=($(find ${RAWDATADIR} -maxdepth 1 -name ${SAMPLEIDS[${SLURM_ARRAY_TASK_ID}]}'*'))
+    toProcess=($(find "${dirs[@]}" -maxdepth 1 -name ${SAMPLEIDS[${SLURM_ARRAY_TASK_ID}]}'*'))
+    ## sort the toProcess array so that R1 and R2 are consecutive 
+    IFS=$'\n' # need to set this as \n rather than default - a space, \t and then \n - so that elements are expanded using \n as delimiter
+    toProcess=($(sort <<<"${toProcess[*]}")) ## sort so that the first element is R1
+    unset IFS 
+    echo ${toProcess}
+    echo "R1 file found is: " $( basename ${toProcess[0]} )
+    
+    echo "Current sample: " ${sampleID} 
+    
+    ## If only $1 is specified, all steps will be run
+    ## option FASTQC: run sequencing QC on raw reads files        
+    if [ $# == 1 ] || [[ $2 =~ 'FASTQC' ]]
+    then
+      
+        echo "Running step 1.1 of ATAC-seq pipeline: FASTQC"
+        module load FastQC 
+        cd ${SCRIPTDIR}
+        sh ./preScripts/fastqc.sh ${sampleID} ${toProcess[0]} ${toProcess[1]}  
+    fi
+    
+    ## option TRIM: run trimming on raw reads files and output trimmed reads.
+    if [ $# == 1 ] || [[ $2 =~ 'TRIM' ]]
+    then
+        module purge
+        module load fastp
+        
+    	  echo "Running step 1.2 of ATAC-seq pipeline: TRIM"
+        cd ${SCRIPTDIR}
+        sh ./preScripts/fastp.sh ${sampleID} ${toProcess[0]} ${toProcess[1]} 
+    fi
+
+    ## option ALIGN: run alignment and post processing on sample
+	  if [ $# == 1 ] || [[ $2 =~ 'ALIGN' ]]
+  	then
+   
+  		echo "Running step 1.3 of ATAC-seq pipeline: ALIGN"
+  		module purge ## had conflict issues if this wasn't run first
+  		module load Bowtie2/2.3.4.2-foss-2018b
+  		module load SAMtools
+  		module load picard/2.6.0-Java-1.8.0_131
+  		export PATH="$PATH:/lustre/projects/Research_Project-MRC190311/software/atac_dnase_pipelines/utils/"
+  		
+  		cd ${SCRIPTDIR}
+  		sh ./ATACSeq/preprocessing/alignment.sh ${sampleID}
+  	fi
+   
+    ## option ENCODE: calculate ENCODEQC metrics on aligned samples
+  	if [ $# == 1 ] || [[ $2 =~ 'ENCODE' ]]
+  	then
+      echo "Running step 1.4 of ATAC-seq pipeline: ENCODE"
+  		module purge
+  		module load SAMtools
+  		module load BEDTools/2.27.1-foss-2018b ##necessary to specify earlier BEDTools version
+          
+      ## load conda env for samstats
+      module load Anaconda3
+      source activate encodeqc
```

Comment:
Would you mind updating this to `conda activate`?
`source activate` is the old way of activating a conda environment and is generally slower.
Also, is there any explanation of how to setup this conda environment in associated documentation?